pkg/services/ngalert/api/api_provisioning.go

```diff
@@ -333,6 +337,15 @@ func (srv *ProvisioningSrv) RoutePostAlertRule(c *contextmodel.ReqContext, ar de
 	if err != nil {
 		return ErrResp(http.StatusBadRequest, err, "")
 	}
+
+	if upstreamModel.IsRecordingRule() && !srv.featureManager.IsEnabledGlobally(featuremgmt.FlagGrafanaManagedRecordingRules) {
+		return ErrResp(
+			http.StatusBadRequest,
+			fmt.Errorf("%w: recording rules cannot be created on this instance", alerting_models.ErrAlertRuleFailedValidation),
+			"",
+		)
+	}
+
 	provenance := determineProvenance(c)
 	createdAlertRule, err := srv.alertRules.CreateAlertRule(c.Req.Context(), c.SignedInUser, upstreamModel, alerting_models.Provenance(provenance))
```

Comment:
Sadly I think we'll need to duplicate much of the validation code, inside the AlertRuleService too. The main API validation can't be reused by provisioning because of how that whole validation flow is architected (it mixes type conversions, that are specific to the main API, alongside validation).

Reply:
This can ofc be saved for a future PR - but if we decide to do so, let's make sure we add a checkbox to the tasklist to remember it. We'll at least need to copy some validation logic, at most need to separate validation in a way that's reusable.